Source/Core/VideoBackends/Vulkan/VKTexture.cpp

```diff
@@ -113,23 +113,17 @@ void VKTexture::Bind(unsigned int stage)
   StateTracker::GetInstance()->SetTexture(stage, m_texture->GetView());
 }
 
-bool VKTexture::Save(const std::string& filename, unsigned int level)
+std::optional<AbstractTexture::RawTextureInfo> VKTexture::MapFullImpl()
 {
-  _assert_(level < m_config.levels);
-
-  // We can't dump compressed textures currently (it would mean drawing them to a RGBA8
-  // framebuffer, and saving that). TextureCache does not call Save for custom textures
-  // anyway, so this is fine for now.
-  _assert_(m_config.format == AbstractTextureFormat::RGBA8);
-
-  // Determine dimensions of image we want to save.
-  u32 level_width = std::max(1u, m_config.width >> level);
-  u32 level_height = std::max(1u, m_config.height >> level);
+  // No support for optimization of full copy
+  return MapRegionImpl(0, 0, 0, m_config.width, m_config.height);
+}
 
-  // Use a temporary staging texture for the download. Certainly not optimal,
-  // but since we have to idle the GPU anyway it doesn't really matter.
-  std::unique_ptr<StagingTexture2D> staging_texture = StagingTexture2D::Create(
-      STAGING_BUFFER_TYPE_READBACK, level_width, level_height, TEXTURECACHE_TEXTURE_FORMAT);
+std::optional<AbstractTexture::RawTextureInfo> VKTexture::MapRegionImpl(u32 level, u32 x, u32 y,
+                                                                        u32 width, u32 height)
+{
+  m_staging_texture = StagingTexture2D::Create(STAGING_BUFFER_TYPE_READBACK, width, height,
```

Comment:
Allocating copy on readback again, this will be slow. Vulkan lets us use persistent mapped buffers, we should take advantage of that.

Also, having two code paths (one for EFB copy readback, and one for general readback) with different performance characteristics is confusing and duplicated..

Reply:
> Allocating copy on readback again, this will be slow. Vulkan lets us use persistent mapped buffers, we should take advantage of that.

Happy to look at this but I'm quite out of the loop of graphics stuff.  It will likely take quite a while.

> Also, having two code paths (one for EFB copy readback, and one for general readback) with different performance characteristics is confusing and duplicated..

The hope on my part was that we move to one readback path for EFB/XFBs.  That was going to be a later PR to switch overall the EFB pokes and such to use this capability.